server/sonar-server/src/test/java/org/sonar/server/permission/ws/RemoveGroupActionTest.java

```diff
@@ -305,4 +307,87 @@ public void removing_project_permission_is_allowed_to_project_administrators() t
 
     assertThat(db.users().selectGroupPermissions(aGroup, project)).containsOnly(CODEVIEWER);
   }
+
+  @Test
+  public void no_effect_when_removing_any_permission_from_group_AnyOne_on_a_private_project() {
```

Comment:
I would prefer to rename `no_effect_...` to `should_not_fail_...`, because it does not contain asserts. (or to add asserts, of course)